core/app/models/spree/store_credit.rb

```diff
@@ -288,4 +289,11 @@ def associate_credit_type
       self.credit_type = Spree::StoreCreditType.find_by_name(credit_type_name)
     end
   end
+
+  def apply_to_orders_in_confirm_state
```

Comment:
`restart_checkout_flow` already calls `.next!` on the order. I kind of wish it did an advance itself, though I'm not sure introducing that change here is the best way to go. So perhaps we'd have to advance the order ourselves?